core/lib/spree/testing_support/capybara_ext.rb

```diff
@@ -85,6 +85,16 @@ def wait_for_ajax(delay = Capybara.default_max_wait_time)
     end
   end
 
+  def wait_for_condition(delay = Capybara.default_max_wait_time)
+    counter = 0
+    delay_threshold = delay * 10
+    until yield
+      counter += 1
+      sleep(0.1)
+      raise "Could not achieve condition within #{delay} seconds." if counter >= delay_threshold
```

Comment:
@chhga can you add this to i18n ?

Reply:
not sure if we want to use i18n for internal specs exceptions.

Reply:
oh, yes, u have right 😄